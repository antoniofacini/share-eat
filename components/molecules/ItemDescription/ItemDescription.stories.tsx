import { Story, Meta } from '@storybook/react/types-6-0'
import ItemDescription from '.'

export default {
  title: 'Molecules/ItemDescription',
  component: ItemDescription,
} as Meta

export const Default: Story = (args) => <ItemDescription {...args} />
Default.args = {}
