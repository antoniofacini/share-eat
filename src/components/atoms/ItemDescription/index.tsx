interface IItemDescription {
  name?: string
  price?: string
  description?: string
  currency?: string
}

const ItemDescription = ({
  name = 'X Bacon',
  price = '20,50',
  description = 'Cat ipsum dolor sit amet, caticus cuteicus bathe private parts with tongue then lick',
  currency = 'R$',
}: IItemDescription) => (
  <div className="bg-gray-dark p-5 rounded-md text-white border-l-6 border-yellow inline-grid grid-cols-3 gap-2">
    <p>{name}</p>
    <p className="col-start-3">
      {currency} {price}
    </p>
    <p className="col-span-3">{description}</p>
  </div>
)

export default ItemDescription
