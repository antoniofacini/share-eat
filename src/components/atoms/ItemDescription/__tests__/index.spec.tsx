import { render } from '@testing-library/react'
import ItemDescription from '..'

describe('<ItemDescription />', () => {
  it('should render ItemDescription component', () => {
    const wrapper = render(<ItemDescription />)
    expect(wrapper).toMatchSnapshot()
  })
})
