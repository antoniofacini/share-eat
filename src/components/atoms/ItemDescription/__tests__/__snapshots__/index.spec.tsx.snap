// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemDescription /> should render ItemDescription component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-gray-dark p-5 rounded-md text-white border-l-6 border-yellow inline-grid grid-cols-3 gap-2"
      >
        <p
          class="font-bold col-span-2"
        >
          X Bacon
        </p>
        <p
          class="col-start-3 font-bold text-end"
        >
          R$
           
          20,50
        </p>
        <p
          class="col-span-3 font-light"
        >
          Cat ipsum dolor sit amet, caticus cuteicus bathe private parts with tongue then lick
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-gray-dark p-5 rounded-md text-white border-l-6 border-yellow inline-grid grid-cols-3 gap-2"
    >
      <p
        class="font-bold col-span-2"
      >
        X Bacon
      </p>
      <p
        class="col-start-3 font-bold text-end"
      >
        R$
         
        20,50
      </p>
      <p
        class="col-span-3 font-light"
      >
        Cat ipsum dolor sit amet, caticus cuteicus bathe private parts with tongue then lick
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
